runtime/src/iree-amd-aie/aie_runtime/iree_aie_runtime.h

```diff
@@ -60,6 +60,97 @@ struct TileLoc {
   bool operator!=(const TileLoc& rhs) const { return !(*this == rhs); }
 };
 
+struct Port {
+  StrmSwPortType bundle;
+  int channel;
+
+  bool operator==(const Port& rhs) const {
+    return std::tie(bundle, channel) == std::tie(rhs.bundle, rhs.channel);
+  }
+
+  bool operator!=(const Port& rhs) const { return !(*this == rhs); }
+
+  bool operator<(const Port& rhs) const {
+    return std::tie(bundle, channel) < std::tie(rhs.bundle, rhs.channel);
+  }
+};
+
+struct DMAChannel {
+  DMAChannelDir direction;
+  int channel;
+
+  bool operator==(const DMAChannel& rhs) const {
+    return std::tie(direction, channel) == std::tie(rhs.direction, rhs.channel);
+  }
+};
+
+struct Switchbox : TileLoc {
+  // Necessary for initializer construction?
+  Switchbox(TileLoc t) : TileLoc(t) {}
+  Switchbox(int col, int row) : TileLoc{col, row} {}
+  bool operator==(const Switchbox& rhs) const {
+    return static_cast<TileLoc>(*this) == rhs;
+  }
+};
+
+struct Connect {
+  Switchbox sb;
+  Port src;
+  Port dst;
+
+  bool operator==(const Connect& rhs) const {
+    return std::tie(src, dst) == std::tie(rhs.src, rhs.dst);
+  }
+};
+
+struct Channel {
```

Comment:
What's the difference between `Channel` and `DMAChannel`. This can be confusing.

Reply:
Channel is a connection between switchboxes, DMAChannel is a dma port (ie direction plus index). I don't remember if these were the names prior - feel free to suggest new names.

Reply:
`SwitchBoxConnection`, `SwitchConnection`?